src/common/appId.js

```diff
@@ -35,3 +26,69 @@ export const oauthLogin = (done = () => 0) => {
         done();
     }
 };
+
+export const getCustomEndpoint = () => ({
+    url  : getStorage('config.server_url'),
+    appId: getStorage('config.app_id'),
+});
+
+export const getDefaultEndpoint = () => ({
+    url  : 'frontend.binaryws.com',
+    appId: getStorage('config.default_app_id') || 1169,
+});
+
+export const getServerAddressFallback = () => getCustomEndpoint().url || getDefaultEndpoint().url;
+
+export const getAppIdFallback = () => getCustomEndpoint().appId || getDefaultEndpoint().appId;
+
+export const getWebSocketURL = () => `wss://${getServerAddressFallback()}/websockets/v3`;
+
+export const generateWebSocketURL = serverUrl => `wss://${serverUrl}/websockets/v3`;
+
+export const getOAuthURL = () =>
+    `https://${getServerAddressFallback()}/oauth2/authorize?app_id=${getAppIdFallback()}&l=${getLanguage().toUpperCase()}`;
+
+const options = {
+    apiUrl   : getWebSocketURL(),
+    websocket: typeof WebSocket === 'undefined' ? require('ws') : undefined, // eslint-disable-line global-require
+    language : getStorage('lang') || 'en',
+    appId    : getAppIdFallback(),
+};
+
+export const generateLiveApiInstance = () => new LiveApi(options);
+
+export const generateTestLiveApiInstance = overrideOptions => new LiveApi(Object.assign(options, overrideOptions));
```

Comment:
This changes contents of `options`. It's usually safer to use the spread syntax:

```
new LiveApi({...options, ... overrideOptions});
```